src/transformers/utils/hub.py

```diff
@@ -1123,6 +1124,31 @@ def move_cache(cache_dir=None, new_cache_dir=None, token=None):
         )
 
 
+class PushInProgress:
+    """
+    Internal class to keep track of a push in progress (which might contain multiple `Future` jobs).
+    """
+
+    def __init__(self, jobs=None):
+        self.jobs = [] if jobs is None else jobs
+
+    def is_done(self):
+        return all(job.done() for job in self.jobs)
+
+    def wait_until_done(self):
+        futures.wait(self.jobs)
+
+    def cancel(self):
+        new_jobs = []
+        for job in self.jobs:
+            if job.running():
+                new_jobs.append(job)
+            elif not job.done():
+                job.cancel()
+
+        self.jobs = new_jobs
```

Comment:
```suggestion
    def cancel(self) -> None:
        for job in self.jobs:
            job.cancel()
```

Except if there is a reason I'm missing, I would not modify the list of jobs here. It adds quite some complexity to understand the for+if+elif statements with no noticeable gain. 

Or maybe something like this that would at least ensure that [`.cancel`](https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Future.cancel) is called on each and every job and let to Python the logic of `if running ... elif not done ...` 
```py
    def cancel(self) -> None:
        self.jobs = [
            job
            for job in self.jobs
            # Cancel the job if it wasn't started yet and remove cancelled/done jobs from the list
            if not (job.cancel() or job.done())
        ]
```

Reply:
I fond your alternative much less clearer than the explicit for loop :sweat_smile: . Also the goal is to not cancel the job that is currently running since it's not really possible to cancel it from what you told me.

Reply:
It's not possible but it doesn't fail either. Actually `future.cancel()` returns a boolean whether the future has been cancelled are not. So the if/else logic is done by Python internally

Reply:
Ok, switched the code to your suggestion :-)